### PR TITLE
Item styling tff 109

### PR DESCRIPTION
**Pull request by tondef001**
### Summary

Styled the list of items and the Student Panel.  Changed the app background to a light grey, Formatted the text of the student panel, and styled each of the items a student has checked out.  The Items will state whether or not they are overdue, as well as provide links to the item page for that item.

### Checklist

- [x] Did you run `npm test`?
- [x] Did you update/add documentation for new methods or changed functionality?

Finally, is it ready to be reviewed and merged: 🎄 

### How to Review

NOTE: Though this branch itself does not rely on it, fully testing this PR will require Boaty McBoatFace.

1. Run this client with dev as the server
2. Enter a student id (123456)
3. Check out some items for the student
4. Inspect the pretty Student panel.
5. Click on the items to travel to the item component for that item
6. Enter a student id of a student with overdue items (111111) (REQUIRES BOATY)
7. The Overdue item has red text and says it's overdue
8. Review the Code/Run the tests

### Links

- [TFF-109](https://msoese.atlassian.net/browse/TFF-109)
Using This PR for Boaty will work until he's in dev: https://github.com/TheFourFifths/consus/pull/36

### CODE REVIEW

File: src/lib/api-client.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ export function createModel(name, description, manufacturer, vendor, location, i
 }
 
 export function searchItem(address) {
-    get('item', {
+    return get('item', {
```
Comment (mattmahn) [inline src/lib/api-client.js]: :+1:

File: src/store/item-store.js [hunk 2]
```diff
@@ -32,4 +36,12 @@ store.registerHandler('ITEMS_RECEIVED', data => {
     items = data.items;
     store.emitChange();
 });
+
+store.registerHandler('STUDENT_FOUND', data => {
+    items = data.items;
+    for(let item of data.items) {
```
Comment (mattmahn) [inline src/store/item-store.js]: such non-functional programming (i.e. not `items.forEach`)

File: src/styles/student.styl [hunk 3]
```diff
@@ -1,15 +1,38 @@
...
-        background-color: darken(#ffffff, 16%)
+        background-color: #d8d8d8
+        border-radius: 3px
+        box-shadow: 0 1px 0 #ccc
+
+        .name, .id
+            margin: 0 0 0 0.5rem
+
+        .equipment-heading, .equipment-none
+            margin 1em 0 0 0.8rem
```
Comment (mattmahn) [inline src/styles/student.styl]: Noice [r]em!

File: src/styles/student.styl [hunk 4]
```diff
@@ -1,15 +1,38 @@
...
+
+        .item-info
+            background: #e5e5e5
+            margin: 0.5em 0.5rem
+            border-radius: 3px
+            padding: 0.3em .8rem
+            box-shadow: 0 1px 0 #ccc
+
+        .item-info:hover
+            background: lighten(#e5e5e5, 15%)
```
Comment (mattmahn) [inline src/styles/student.styl]: Nit: `#e5e5e5` doesn't match any other color. odd.

File: test/unit/store/item-store.js [hunk 5]
```diff
@@ -39,5 +39,21 @@ describe('ItemStore', () => {
...
 
+    it('should find overdue items', () => {
+        Dispatcher.handleAction('STUDENT_FOUND',{
+            id: '432345',
+            name: 'Thor',
+            items: [{address:1, timestamp:0},{address:2},{address:3, timestamp:0},{address:4}]
+        });
+        let items = ItemStore.getAllItems();
+        for(let item of items){
+            if(item.address%2 == 1){
```
Comment (mattmahn) [inline test/unit/store/item-store.js]: suck lack of ` `

File: src/styles/student.styl [hunk 6]
```diff
@@ -1,15 +1,38 @@
...
+            color: red
+
+        .item-info
+            background: #e5e5e5
+            margin: 0.5em 0.5rem
+            border-radius: 3px
+            padding: 0.3em .8rem
+            box-shadow: 0 1px 0 #ccc
+
+        .item-info:hover
```
Comment (mattmahn) [inline src/styles/student.styl]: This should be indented and be `&:hover`

File: src/styles/student.styl (hunk 4 above)
Comment (tondef001) [inline src/styles/student.styl]: It is the original color of .item-info

File: src/store/item-store.js [hunk 7]
```diff
@@ -32,4 +36,12 @@ store.registerHandler('ITEMS_RECEIVED', data => {
...
+        item.isOverdue = store.isOverdue(item);
```
Comment (thofmann) [inline src/store/item-store.js]: Similar to the `student.hasOverdueItem` issue here... We could store `false` here now, but the value would be stale if the item becomes overdue during this time.

File: src/styles/student.styl [hunk 8]
```diff
@@ -1,15 +1,38 @@
...
+
+        .item-info
+            background: #e5e5e5
+            margin: 0.5em 0.5rem
+            border-radius: 3px
+            padding: 0.3em .8rem
+            box-shadow: 0 1px 0 #ccc
+
+            &:hover
+                background: lighten(#e5e5e5, 15%)
```
Comment (thofmann) [inline src/styles/student.styl]: `lighten` :+1:

**Pull request merged**